invenio_app_ils/providers/api.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019-2021 CERN.
+#
+# invenio-app-ils is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Provider API."""
+
+from functools import partial
+
+from invenio_pidstore.models import PIDStatus
+from invenio_pidstore.providers.recordid_v2 import RecordIdProviderV2
+
+from invenio_app_ils.acquisition.proxies import current_ils_acq
+from invenio_app_ils.errors import RecordHasReferencesError
+from invenio_app_ils.fetchers import pid_fetcher
+from invenio_app_ils.ill.proxies import current_ils_ill
+from invenio_app_ils.minters import pid_minter
+from invenio_app_ils.records.api import IlsRecord
+
+PROVIDER_PID_TYPE = "provid"
+PROVIDER_PID_MINTER = "provid"
+PROVIDER_PID_FETCHER = "provid"
+
+ProviderIdProvider = type(
+    "ProviderIdProvider",
+    (RecordIdProviderV2,),
+    dict(pid_type=PROVIDER_PID_TYPE, default_status=PIDStatus.REGISTERED),
+)
+provider_pid_minter = partial(pid_minter, provider_cls=ProviderIdProvider)
+provider_pid_fetcher = partial(pid_fetcher, provider_cls=ProviderIdProvider)
+
+
+class Provider(IlsRecord):
+    """Provider class."""
+
+    _pid_type = PROVIDER_PID_TYPE
+    _schema = "providers/provider-v1.0.0.json"
+
+    def delete(self, **kwargs):
+        """Delete record."""
+        if self["type"] == "VENDOR":
+            search_cls = current_ils_acq.order_search_cls
+            ref_type = "Order"
+        else:
```

Comment:
I would add just to be safe:
```suggestion
        elif ... == '...':
            ...
        else:
            throw Error
```

Reply:
Actually, is it better than to have a `enum` instead of a vocab? WDYT?